File: C++/gnome_sort.cpp
#include <iostream>
using namespace std;

void gnomeSort(int arr[], int n) {
    int index = 0;

    while (index < n) {
        if (index == 0 || arr[index] >= arr[index - 1]) {
            index++;
        } else {
            swap(arr[index], arr[index - 1]);
            index--;
        }
    }
}

void printArray(const int arr[], int n) {
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    int arr[] = {34, 2, 10, -9};
    int n = sizeof(arr) / sizeof(arr[0]);

    gnomeSort(arr, n);

    printArray(arr, n);

    return 0;
}

File: Python/gnome_sort.py
def gnome_sort(arr):
    n = len(arr)
    index = 0

    while index < n:
        if index == 0 or arr[index] >= arr[index - 1]:
            index += 1
        else:
            arr[index], arr[index - 1] = arr[index - 1], arr[index]
            index -= 1

    return arr

arr=[3, 7, 4, 8, 6, 2, 1, 5, 0, -5]
print(gnome_sort(arr))